integration_tests/src/main/python/delta_lake_write_test.py

```diff
@@ -206,30 +271,44 @@ def do_write(spark, path):
         writer.saveAsTable(table)
-    assert_gpu_and_cpu_writes_are_equal_collect(
-        do_write,
-        lambda spark, path: spark.read.format("delta").table(path_to_table[path]),
-        data_path,
-        conf=confs)
+    if not is_spark_353_or_later():
+        assert_gpu_and_cpu_writes_are_equal_collect(
+            do_write,
+            lambda spark, path: spark.read.format("delta").table(path_to_table[path]),
+            data_path,
+            conf=confs)
+    else:
+        assert_gpu_fallback_write(
+            do_write,
+            lambda spark, path: spark.read.format("delta").table(path_to_table[path]),
+            data_path,
+            fallback_classes,
+            conf=confs)
+        pytest.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/12930, https://github.com/NVIDIA/spark-rapids/issues/12931")
 
+@allow_non_gpu_conditional(is_spark_353_or_later(), "AtomicCreateTableAsSelectExec, AppendDataExecV1")
 @allow_non_gpu('DataWritingCommandExec', 'WriteFilesExec', *delta_meta_allow)
 @delta_lake
 @ignore_order(local=True)
 @pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
 @pytest.mark.parametrize("enable_deletion_vectors", deletion_vector_values_with_350DB143_xfail_reasons(
                             enabled_xfail_reason="https://github.com/NVIDIA/spark-rapids/issues/12041"), ids=idfn)
 def test_delta_atomic_create_table_as_select(spark_tmp_table_factory, spark_tmp_path, enable_deletion_vectors):
-    _atomic_write_table_as_select(delta_write_gens, spark_tmp_table_factory, spark_tmp_path, overwrite=False, enable_deletion_vectors=enable_deletion_vectors)
+    _atomic_write_table_as_select(delta_write_gens, spark_tmp_table_factory, spark_tmp_path, overwrite=False, enable_deletion_vectors=enable_deletion_vectors,
+        fallback_classes=["AtomicCreateTableAsSelectExec", "AppendDataExecV1"])
 
+@allow_non_gpu_conditional(is_spark_353_or_later(), "AtomicReplaceTableAsSelectExec, AppendDataExecV1")
 @allow_non_gpu('DataWritingCommandExec', 'WriteFilesExec', *delta_meta_allow)
 @delta_lake
 @ignore_order(local=True)
 @pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")
 @pytest.mark.parametrize("enable_deletion_vectors", deletion_vector_values_with_350DB143_xfail_reasons(
                             enabled_xfail_reason="https://github.com/NVIDIA/spark-rapids/issues/12041"), ids=idfn)
 def test_delta_atomic_replace_table_as_select(spark_tmp_table_factory, spark_tmp_path, enable_deletion_vectors):
-    _atomic_write_table_as_select(delta_write_gens, spark_tmp_table_factory, spark_tmp_path, overwrite=True, enable_deletion_vectors=enable_deletion_vectors)
+    _atomic_write_table_as_select(delta_write_gens, spark_tmp_table_factory, spark_tmp_path, overwrite=True, enable_deletion_vectors=enable_deletion_vectors,
+        fallback_classes=["AtomicReplaceTableAsSelectExec", "AppendDataExecV1"])
```

Comment:
This test will xfail with the reason = `"https://github.com/NVIDIA/spark-rapids/issues/12930, https://github.com/NVIDIA/spark-rapids/issues/12931"`. But the description of either issues do not cover supporting `AtomicReplaceTableAsSelectExec`. I presume though that #12931 can cover it. If that's the case, can you update the issue description? If not, please add a new issue and link it here.